drivers/soundwire/bus.c

```diff
@@ -1749,8 +1749,11 @@ int sdw_handle_slave_status(struct sdw_bus *bus,
 			continue;
 
 		if (status[i] == SDW_SLAVE_UNATTACHED &&
-		    slave->status != SDW_SLAVE_UNATTACHED)
+		    slave->status != SDW_SLAVE_UNATTACHED) {
+			dev_warn(&slave->dev, "Slave %d state check1: UNATTACHED, status was %d\n",
```

Comment:
I'm a bit confused about the warning @plbossart. In both cases it looks like modifying the slave status to UNATTACHED status is intentional. Why should it be a warning then?

Reply:
@plbossart Maybe a dev_info() instead? Would still be visible in dmesg (and seen in bug reports).

Reply:
Slave will not be unattached for no reason. Warning is fine for me. Of cause dev_info is also fine for me.

Reply:
the fact that the 'Slave' device is no longer synchronized and attached AFTER the enumeration is not expected. Worse, in such cases the device doesn't seem to come back and remains unattached, so I think dev_warn is exactly what we want.
dev_info() does not carry the weight of 'this is not normal'.
dev_err() would be too strong, we've seen in some cases that devices come and go, we can't flag an error in the common bus code for such a behavior.